PublicMapping/districtbuilder
# Send email via SES

## Overview

Creates SES verification and DKIM records in AWS and configures the application to use SES for email transport.

### Notes

~I've opened up an AWS support case to have SES taken out of sandbox mode so that we can send registration emails to arbitrary recipients. I'll indicate when the ticket goes through in a comment.~

Our SES sandbox restrictions have been removed. Our new SES email limit is 50,000 per day at a rate of 14 per second.

~The verification link in the registration email will not be valid. I set up #163 to track this bug.~

## Testing Instructions

- Run `docker-compose -f docker-compose.ci.yml run --rm terraform`.
- Run `GIT_COMMIT=d3ad718 ./scripts/infra plan`. Confirm that there are no new changes to be applied.
- Log in to the DistrictBuilder AWS Console.
- Go to https://staging.districtbuilder.azavea.com/register and register a new user.
- Confirm that you receive a verification email from `no-reply@staging.districtbuilder.azavea.com`.
- Start up a local development server with `scripts/setup` and `scripts/server`.
- Go to the local `server` running on `http://localhost:3005`.
- Perform the same registration process.
- Check the `server` console output and check that an email from `no-reply@staging.districtbuilder.azavea.com` was logged.

Closes #54
